recirq/quantum_chess/circuit_transformer.py

```diff
@@ -400,48 +415,52 @@ def transform(self, circuit: cirq.Circuit) -> cirq.Circuit:
         return cirq.Circuit(updater.add_swaps())
 
 
-class SycamoreDecomposer(cirq.PointOptimizer):
-    """Optimizer that decomposes all three qubit operations into
+def transform_op_into_sqrt_iswap(op: cirq.Operation, index: int) -> cirq.OP_TREE:
+    if len(op.qubits) > 3:
+        raise DeviceMappingError(f"Four qubit ops not yet supported: {op}")
+    if op.gate == cirq.SWAP or op.gate == cirq.CNOT or op.gate == cirq.TOFFOLI:
+        return cg.optimized_for_sycamore(cirq.Circuit(op)).all_operations()
+    elif isinstance(op, cirq.ControlledOperation):
+        if not all(v == 1 for values in op.control_values for v in values):
+            raise DeviceMappingError(f"0-controlled ops not yet supported: {op}")
+        qubits = op.sub_operation.qubits
+        if op.gate.sub_gate == cirq.ISWAP:
+            return controlled_iswap.controlled_iswap(*qubits, *op.controls)
+        if op.gate.sub_gate == cirq.ISWAP**-1:
+            return controlled_iswap.controlled_iswap(
+                *qubits, *op.controls, inverse=True
+            )
+        if op.gate.sub_gate == cirq.ISWAP**0.5:
+            return controlled_iswap.controlled_sqrt_iswap(*qubits, *op.controls)
+        if op.gate.sub_gate == cirq.ISWAP**-0.5:
+            return controlled_iswap.controlled_inv_sqrt_iswap(*qubits, *op.controls)
+        if op.gate.sub_gate == cirq.X:
+            if len(op.qubits) == 2:
+                return cg.optimized_for_sycamore(
+                    cirq.Circuit(cirq.CNOT(*op.controls, *qubits))
+                ).all_operations()
+            if len(op.qubits) == 3:
+                return cg.optimized_for_sycamore(
+                    cirq.Circuit(cirq.TOFFOLI(*op.controls, *qubits))
+                ).all_operations()
+    return op
+
+
+@cirq.transformer
+def decompose_into_sqrt_iswap(
```

Comment:
This entire transformer can now be replaced by a single function call in Cirq:
```python
cirq.optimize_for_target_gateset(circuit, gateset=cirq.SqrtIswapTargetGateset(), ignore_failures=False)
```

As an evidence, here is a colab which tests compilation of all the different cases hardcoded in the `transform_op_into_sqrt_iswap` method above:

https://colab.research.google.com/drive/1sjacBgm1oqa2giTQ_QUlu_GxKqJ9GLsR?resourcekey=0-UIKWzHrpTw58dWE-VPnqlw&usp=sharing